lock

```diff
@@ -1,70 +1,90 @@
+if [ -e $OUTPUT_IMG ]; then
+	$LOCK_CMD
+	exit 0
 fi
 
-#Execute xrandr to get information about the monitors:
-while read LINE
-do
-  #If we are reading the line that contains the position information:
-  if [[ $LINE =~ $DISPLAY_RE ]]; then
-    #Extract information and append some parameters to the ones that will be given to ImageMagick:
-    SCREEN_WIDTH=${BASH_REMATCH[1]}
-    SCREEN_HEIGHT=${BASH_REMATCH[2]}
-    SCREEN_X=${BASH_REMATCH[3]}
-    SCREEN_Y=${BASH_REMATCH[4]}
+# Execute xrandr to get information about the monitors:
+#while read LINE
+#do
+while IFS= read -r LINE; do
+	# If we are reading the line that contains the position information:
+	if [[ $LINE =~ $DISPLAY_RE ]]; then
+		# Extract information and append some parameters to the ones that will be given to ImageMagick:
+		SCREEN_WIDTH=${BASH_REMATCH[1]}
+		SCREEN_HEIGHT=${BASH_REMATCH[2]}
+		SCREEN_X=${BASH_REMATCH[3]}
+		SCREEN_Y=${BASH_REMATCH[4]}
     
-    CACHE_IMG="$CACHE_FOLDER""$SCREEN_WIDTH"x"$SCREEN_HEIGHT"."$MD5_BKG_IMG"."$MD5_TEXT_IMG".png
-    ## if cache for that screensize doesnt exist
-    if ! [ -e $CACHE_IMG ]
-    then
-	# if this is the first time cache was not hit then
-	if ! [ $(echo $TEXT_IMG_INFO) ]
-	then	
-	    #Get dimensions of the text image
-	    TEXT_IMG_INFO=`identify $TEXT_IMG`
-	    [[ $TEXT_IMG_INFO =~ $IMAGE_RE ]]
-	    TEXT_IMG_WIDTH=${BASH_REMATCH[1]}
-	    TEXT_IMG_HEIGHT=${BASH_REMATCH[2]}
-        fi
-	# Create image for that screensize
-        TEXT_X=$(($X+$SCREEN_WIDTH/2-$TEXT_IMG_WIDTH/2))
-        TEXT_Y=$(($Y+$SCREEN_HEIGHT/2-$TEXT_IMG_HEIGHT/2+200))
-        eval convert '\(' '$BKG_IMG' '-resize' '${SCREEN_WIDTH}X${SCREEN_HEIGHT}!' '\)' \
-		'\(' '$TEXT_IMG' '-geometry' '+$TEXT_X+$TEXT_Y' '\)' '-composite' '$CACHE_IMG'
-    fi
+		CACHE_IMG="$CACHE_FOLDER""$SCREEN_WIDTH"x"$SCREEN_HEIGHT"."$MD5_BKG_IMG".png
+		# If cache for that screensize doesnt exist
+		if ! [ -e $CACHE_IMG ]; then
+			#eval convert '$BKG_IMG' '-resize' '${SCREEN_WIDTH}X${SCREEN_HEIGHT}!' '$CACHE_IMG'
+			eval convert '$BKG_IMG' '-resize' '${SCREEN_WIDTH}X${SCREEN_HEIGHT}^' '-gravity' 'Center' '-crop' '${SCREEN_WIDTH}X${SCREEN_HEIGHT}+0+0' '+repage' '$CACHE_IMG'
+		fi
 
-    # Decide size of output image
-    if (( $OUTPUT_IMG_WIDTH < $SCREEN_WIDTH+$SCREEN_X )); then OUTPUT_IMG_WIDTH=$(($SCREEN_WIDTH+$SCREEN_X)); fi;
-    if (( $OUTPUT_IMG_HEIGHT < $SCREEN_HEIGHT+$SCREEN_Y )); then OUTPUT_IMG_HEIGHT=$(( $SCREEN_HEIGHT+$SCREEN_Y )); fi;
+		# Decide size of output image
+		if (( $OUTPUT_IMG_WIDTH < $SCREEN_WIDTH+$SCREEN_X )); then OUTPUT_IMG_WIDTH=$(($SCREEN_WIDTH+$SCREEN_X)); fi;
+		if (( $OUTPUT_IMG_HEIGHT < $SCREEN_HEIGHT+$SCREEN_Y )); then OUTPUT_IMG_HEIGHT=$(( $SCREEN_HEIGHT+$SCREEN_Y )); fi;
 
-    PARAMS="$PARAMS $CACHE_IMG -geometry +$SCREEN_X+$SCREEN_Y -composite "
-  fi
-done <<<"`xrandr`"
+		PARAMS="$PARAMS $CACHE_IMG -geometry +$SCREEN_X+$SCREEN_Y -composite "
+	fi
+done <<<"$(echo "`xrandr | grep -Po $DISPLAY_RE`")"
```

Comment:
don't want to add another dependency `grep`. you can use it in your fork master.

Reply:
Should we really call `grep` a "dependency"? You might have a point but I imagined to myself that there'll be no machines to use your script without `grep` installed.

Reply:
I'm on arch linux which doesn't come with grep installed. I suppose arch Linux is a special case. You are right, most distros do have grep included. But its still a dependency. Besides regex with bash is working fine so no reason using grep for something that can be done in plain bash.